test/e2e/areas/action-bar/editor-action-bar.test.ts

```diff
@@ -152,3 +127,66 @@ async function verifyLineNumbersVisibility(page, isVisible: boolean) {
 		isVisible ? await lineNumbers.toBeVisible() : await lineNumbers.not.toBeVisible();
 	}
 }
+
+async function verifyOpenChanges(page: Page) {
+	await test.step('verify "open changes" shows diff', async () => {
+
+
+		// make change & save
+		await page.getByText('date', { exact: true }).click();
+		await page.keyboard.press('X');
+		await bindPlatformHotkey(page, 'S');
+
+		// click open changes & verify
+		await page.getByLabel('Open Changes').nth(1).click();
+		await expect(page.getByLabel('Revert Block')).toBeVisible();
+		await expect(page.getByLabel('Stage Block')).toBeVisible();
+		await page.getByRole('tab', { name: 'quarto_basic.qmd (Working' }).getByLabel('Close').click();
+
+		// undo changes & save
+		await bindPlatformHotkey(page, 'Z');
+		await bindPlatformHotkey(page, 'S');
+	});
+}
+
+async function bindPlatformHotkey(page: Page, key: string) {
+	await page.keyboard.press(process.platform === 'darwin' ? `Meta+${key}` : `Control+${key}`);
+}
+
+async function verifyOpenViewerRendersHtml(app: Application) {
+	await test.step('verify "open in viewer" renders html', async () => {
+		await app.code.driver.page.getByLabel('Open in Viewer').nth(1).click();
+		const viewerFrame = app.code.driver.page.locator('iframe.webview').contentFrame().locator('#active-frame').contentFrame();
+		const cellLocator = app.web
```

Comment:
should some of this be in positronViewer.  It seems like there is a bit of duplication

Reply:
Or would it be PositronEditor? Maybe both, I guess I'll find out...

Reply:
yeah, agreed about verifications in the POM.  we definitely have some but not my favorite.  was thinking more about locators that might be reusable